src/lib/datepicker/datepicker.ts

```diff
@@ -445,7 +445,7 @@ export class MatDatepicker<D> implements OnDestroy, CanColor {
       .withTransformOriginOn('.mat-datepicker-content')
       .withFlexibleDimensions(false)
       .withViewportMargin(8)
-      .withPush(false)
+      .withPush(true)
```

Comment:
This line can be removed altogether since pushing is enabled by default. Also this will also keep the datepicker in the viewport if the user tries to scroll away, which we probably don't want. You can enable pushing only for the first positioning by enabling locking using `withLockedPosition()`.

@mmalerba is this something we want enabled on the datepicker?

Reply:
@crisbeto yep, it looks kinda broken on small screens without this.